office_space_allocation/amity.py

```diff
@@ -0,0 +1,150 @@
+            if r == room:
+                return True
+        return False
+
+    def find_room(self, name):
+        """
+        Find a ```Room``` object using name
+        :param name: ```str``` name of the Room to look for
+        :return: ```Room``` object if found
+        """
+        name = name.lower()
+        for r in self.all_rooms:
+            if name == r.get_name().lower():
+                return r
+        raise ValueError(name + " not found!")
+
+    def allocate_room(self, person):
+        """
+        Allocates a room to ```Person``` person.
+
+        This is done randomly.
+        :param person: ```Person``` to allocate room
+        :return: room : ```Room``` allocated , if possible
+        """
+        if isinstance(person, Staff):
+            # allocate only offices
+            offices = [of for of in self.all_rooms if isinstance(of, Office)]
+            if len(offices) < 1:
+                raise IndexError("There are no more Office rooms to allocate. Please create some rooms before making "
+                                 "allocations")
+            else:
+                rm = random.choice(offices)
+
+                rm.add_person(person)
+                self.allocated_rooms.append(rm)
+                return rm
+        elif isinstance(person, Fellow):
+            # raise exception if there are no rooms
+            if len(self.all_rooms) < 1:
+                raise IndexError("There are no available rooms. Please create some rooms before making allocations.")
+            else:
+                office_rm = random.choice(self.all_rooms)
+
+                office_rm.add_person(person)
+                self.allocated_rooms.append(office_rm)
+                return office_rm
+        else:
+            raise TypeError("Person argument must be of type Staff or Fellow")
+
+    def find_person(self, name):
+        """
+        Finds a Person using name.
+        Name can be the full name, first name, last name or part of either
+        :param name:
+        :return: a tuple with Person(s) objects found in the list if persons
+        """
+        name = name.strip()
+        res = [p for p in self.all_persons if name.title() in p.get_full_name()]
+        if not res:
+            raise Exception("Person was not found in the system")
+        else:
+            return tuple(res)
+
+    def reallocate_person(self, person_name, room_name):
+        """
+        Reallocates `Person` to another `Room` if possible
+
+        :param: person_name: Name of the `Person`
+        :param: room_name: Name of the `Room`
+        :return: a tuple with (`Person`, `Room`) if reallocation is successful
+        """
+        person_name = person_name.strip()
+        room_name = room_name.strip()
+
+        try:
+            # only one person should be found
+            people_found = self.find_person(person_name)
+            if len(people_found) > 1:
+                raise MultiplePeopleFoundException
```

Comment:
Why pass if you encounter a MultiplePeopleFoundException yet only one person should be found?
Peharps you should return something